subsys/dfu/dfu_target/src/dfu_target.c

```diff
@@ -37,10 +37,10 @@ LOG_MODULE_REGISTER(dfu_target, CONFIG_DFU_TARGET_LOG_LEVEL);
 static const struct dfu_target *current_target;
 static int current_img_num = -1;
 
-int dfu_target_img_type(const void *const buf, size_t len)
+enum dfu_target_image_type dfu_target_img_type(const void *const buf, size_t len)
 {
 	if (len < MIN_SIZE_IDENTIFY_BUF) {
-		return -EAGAIN;
+		return DFU_TARGET_IMAGE_TYPE_NONE;
```

Comment:
This does not indicate that an error occurred, should we instead return an `int err` and then have an output pointer for the image type?

Reply:
Technically it is not an error.
It is just that the given buffer does not contain any image  that we could identify.

I would like to keep this as a enum for two reasons:
* Strict typing is usually better for error checks.
* I can confidently use the returned value in bitmask operations and get the correct result.

Only users of this API I can spot of are not using the EAGAIN error, so I seems meaningless to return it.